sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImpl.java

```diff
@@ -471,4 +471,83 @@ private void waitForFuturesToFinish() throws ExecutionException, InterruptedExce
   public Writer createWriter(CassandraIO.Write<T> spec) {
     return new WriterImpl(spec);
   }
+
+  /** Deleter storing an entity into Apache Cassandra database. */
+  protected class DeleterImpl implements Deleter<T> {
+    /**
+     * The threshold of 100 concurrent async queries is a heuristic commonly used by the Apache
+     * Cassandra community. There is no real gain to expect in tuning this value.
+     */
+    private static final int CONCURRENT_ASYNC_QUERIES = 100;
+
+    private final CassandraIO.Delete<T> spec;
+
+    private final Cluster cluster;
+    private final Session session;
+    private final MappingManager mappingManager;
+    private List<ListenableFuture<Void>> deleteFutures;
+
+    DeleterImpl(CassandraIO.Delete<T> spec) {
+      this.spec = spec;
+      this.cluster =
+          getCluster(
+              spec.hosts(),
+              spec.port(),
+              spec.username(),
+              spec.password(),
+              spec.localDc(),
+              spec.consistencyLevel());
+      this.session = cluster.connect(spec.keyspace());
+      this.mappingManager = new MappingManager(session);
+      this.deleteFutures = new ArrayList<>();
+    }
+
+    /**
+     * Delete the entity to the Cassandra instance, using {@link Mapper} obtained with the {@link
+     * MappingManager}. This method uses {@link Mapper#deleteAsync(Object)} method, which is
+     * asynchronous. Beam will wait for all futures to complete, to guarantee all writes have
+     * succeeded.
+     */
+    @Override
+    public void delete(T entity) throws ExecutionException, InterruptedException {
+      Mapper<T> mapper = (Mapper<T>) mappingManager.mapper(entity.getClass());
+      this.deleteFutures.add(mapper.deleteAsync(entity));
```

Comment:
Similar to the previous comment. Do you think we can make current Writer somehow extensible for this case? I have the impression this line is the only difference between the write and the delete, or I am missing something else?